docs_src/design/adr/0019-Service-List.md

```diff
@@ -0,0 +1,92 @@
+  - There is a question about who or what service would put this file into the file system or volume mount – especially since security bootstrapping would need it first but this file would be needed in non-secure runtime as well.
+  - There isn’t a central service that would own the file – especially for dev/test.
+  - It has to be a service that is always there (like core metadata), but also available from the start of all services (which metadata is not in the case of security services).  Ideally, this service should not require root user privileaes.
+  - Services would have to monitor the file or volume for changes in order to be able to address the add/removal of services at runtime.
+  - How would the security bootstrapping address the add/removal of services and get new tokens established?  The bootstrapping service isn’t even long running to do this.
+- Have global configuration in Consul (or config/reg service) that everyone reads.
+  - When Consul is used, there would be a single authoritative list of services provided to all requesting services through Consul. Updates to Consul would be transactional (ACID).
+  - When Consul is not used, have a common service list in each service configuration.toml file that everyone reads.  Further, when Consul is not used, each service is using the same means (addressing #1 above) to get its service list (that is going to its config TOML file locally).  However, each service list in each configuration file could be different (leading to service list inconsistencies) and there is not a single, authoritative list when using local configuration (making ACID transactions to the service list per #3 above very difficult – especially if services and their configuration file holding the copy of the service list are distributed across hosts).
+  - Using Consul would make sure that the service list is well formed (meaining adherence to format, but this does not mean that the service list necessarily adheres to any validity rules unless additional code interacting with Consul changes is checking for well formed-ness and validity) and that any change operation is ACID (but only with regard to Consul and the service list - this does not mean the transaction would incorporate additional needs such as the API Gateway updating new routes).
+  - When using the local configuration, some of the same issues exist as listed in the first alternative (validation of the file, etc.).
+  - Services would have to be notified by Consul if the service list changes or monitor the local config for a service list change in order to be able to address the add/removal of services at runtime.
+  - Again, how would the security bootstrapping address the add/removal of services and get new tokens established?  The bootstrapping service isn’t even long running to do this.
+  - We could use Consul for dynamic service needs and when Consul is not used resolve that the service list is static and requires a restart of the services.
```

Comment:
How about a Registry implementation that uses new API on Core Metadata for simple registry. Yes, now Core Metadata has to be up first...

Reply:
Could also add a key-value store API to core-metadata, then all of the "but what if we haven't got a registry / config provider?" code goes away :)

Reply:
added as an alternate